storage/core/Cargo.toml

```diff
@@ -7,6 +7,7 @@ license = "MIT"
 [dependencies]
 
 # Local dependencies
+common = { path = '../../common' }
```

Comment:
I think this is a timebomb waiting to blow up with some circular dependency in the future... I see you're only importing sync primitives from common. Please, either make a separate crate for that, or just use primitives from std directly. I cannot really assess the risk on the latter option, but if you find it risky and can cause big compiler errors and confuse other devs, please go with the former option.

Reply:
The `common::concurrency` module switches what is used between primitives from `std` and from `loom`, depending whether we are compiling for loom or not. Using primitives from `std` directly would mean loom would not do any of the fancy thread interleaving stuff.

Maybe `utils` is a more appropriate place to put the `concurrency` stuff in?

Reply:
That sounds OK, at least for now. Let's move it to utils.

Reply:
#380